compose/material/material/src/desktopMain/kotlin/androidx/compose/material/DesktopAlertDialog.desktop.kt

```diff
@@ -158,7 +158,7 @@ fun AlertDialog(
         ) { modifier ->
             AlertDialogContent(
                 buttons = buttons,
-                modifier = modifier.width(IntrinsicSize.Min),
+                modifier = modifier.width(IntrinsicSize.Max),
```

Comment:
Thanks for the detailed explanation!

> Maybe dialogs could have a default width set, and the developer can override that default if it doesn't work?

I think, it is a good idea. The default behaviour should always provide a solution that works out of the box, preferably works well out of the box.

The new default behavior Is not very pretty for desktop - desktop windows are big, and it isn't right to extend small alert dialog to the whole width, if we have a long text.

What if instead of changing it to Max, we define minWidth?

Reply:
I think our solution should have these two properties:
1. It works well for the typical case where the text is short.
2. It's fully customizable by the user (developer), if he chooses so.
The `IntrinsicSize.Max` solution does have these two properties.

> What if instead of changing it to Max, we define minWidth?
Do you mean `widthIn(min = MIN_ALERT_DIALOG_WIDTH)`? If we do just that, it would make the maximum width unlimited, resulting in dialogs always filling the window (even with short text).
If we do `width(IntrinsicSize.Min).widthIn(min = MIN_ALERT_DIALOG_WIDTH)` I think that would be equivalent to `widthIn(max = MIN_ALERT_DIALOG_WIDTH)` (note the `max`).

Whatever we choose, we'll need to verify that the user can override it to his own decision by passing `width`, `widthIn` or any other width-controlling modifier, to the `AlertDialog`.

Reply:
Deciding on the value of `MIN_ALERT_DIALOG_WIDTH` would be hard. There's a lot of variation in monitor sizes, and it changes over the time. On my 27" `400.dp` looks reasonable, but on a small laptop, it would be quite big. On the other hand, `200.dp` on my monitor would be fairly small.

Reply:
> Deciding on the value

When we decide on something that is a part of some design, it usually should have specifications for the dimensions.

For [material](https://m2.material.io/components/dialogs#specs), it is 280.dp for mobile, and 560.dp for desktop. [material3 ](https://m3.material.io/components/dialogs/specs) doesn't have desktop specs, but have min=280.dp, max=560.dp

I am not sure also, if 560.dp will be acceptable for laptops, but the smallest of them usually have the 1366 x 768 resolution, which is 2.5x greater. Also, the width of alert dialogs should depend on the content more than on the surroundings (they should only restrict).

Besides that, it would be good to have paddings around the dialog (I haven't found specs for them).

> we'll need to verify that the user can override

Material design usually is very opinionated. But we can use common sense here and allow flexibility that won't hurt.
The current material3 AlertDialog, for example, has `sizeIn(minWidth = MinWidth, maxWidth = MaxWidth)`, which will allow to override it with `requiredSizeIn(minWidth = MinWidth, maxWidth = MaxWidth)`. Maybe we can do the same here.

> Deciding on the value of MIN_ALERT_DIALOG_WIDTH would be hard

Probably `defaultMinSize(minWidth=)` will do what we want. Not sure how it works, but my guess if the width isn't specified, it wraps the content, but no less than `minWidth`.

Also, we can consider just `sizeIn(280.dp, 560.dp)` as in material3.

Reply:
> When we decide on something that is a part of some design, it usually should have specifications for the dimensions.

How do we use the values from material or material3, and how do we know which values to use?

> Besides that, it would be good to have paddings around the dialog (I haven't found specs for them).

Agreed.

> Probably defaultMinSize(minWidth=) will do what we want. 

Ok, I'll test that.

Reply:
> Probably defaultMinSize(minWidth=) will do what we want.

It doesn't work. Unless we explicitly limit the width, this causes the dialog to take all the window's width:
```
        buttons = {
            // TODO: move the modifiers to FlowRow when it supports a modifier parameter
            Box(Modifier.fillMaxWidth().padding(all = 8.dp)) {
                AlertDialogFlowRow(
```

Reply:
> How do we use the values from material or material3, and how do we know which values to use?

Reading the specs of this design. They are publicly available on the site I mentioned. Or discuss it with the Jetpack Compose team. If we would design completely a new dialog, we would be free from such restrictions.

The general rule of Compose always was - if `material` doesn't fit your requirements within the material specs, don't use it, use more low-level functionality (that should be easely "composable"), or use another design system. In our case we can provide low-level `Layer` and recommend to use it with `Surface`.

> It doesn't work. Unless we explicitly limit the width

We should also keep `IntrinsicSize.Min` when we use `defaultMinSize`. Does it work with it?

Reply:
> Reading the specs of this design.

No, I mean would it be a hardcoded constant, a field in `MaterialTheme`, or something else? How would we know whether to use the constant for material or material3?

> We should also keep IntrinsicSize.Min when we use defaultMinSize. Does it work with it?

With `modifier.width(IntrinsicSize.Min).defaultMinSize(minWidth = 280.dp)` it becomes impossible to allow it to match the content and be wider than `280.dp`. You can specify an exact number with `width(400.dp)`, but you can't say "let it match the content up to `400.dp` but not larger.

Reply:
> I mean would it be a hardcoded constant

If we look at the other code in material module, the most constants are hardcoded.

> whether to use the constant for material or material3?

material and material3 has own independent AlertDialog's

> You can specify an exact number with width(400.dp)

Let's discuss online tomorrow

Reply:
Discussed that the current variant in the PR is okay.